src/TheThingsNetwork.cpp

```diff
@@ -357,15 +357,21 @@ void TheThingsNetwork::clearReadBuffer()
   }
 }
 
-size_t TheThingsNetwork::readLine(char *buffer, size_t size)
+size_t TheThingsNetwork::readLine(char *buffer, size_t size, uint8_t attempts = 3) // Default timeout value 10s and is set in class initiator
 {
   size_t read = 0;
-  while (read == 0)
+  while (read == 0 && attempts>0)
   {
     read = modemStream->readBytesUntil('\n', buffer, size);
   }
-  buffer[read - 1] = '\0'; // set \r to \0
-  return read;
+  if(attempts<=0){ // If attempts is activated return 0 and set RN state marker
+    this->radioModuleInvalidState = true; // Inform the application about the radio module is not responsive.
+    return 0;
+  }
+  else{
```

Comment:
Please make spacing consistent around braces and brackets. Also no need for `else` after a `return`.

Reply:
Fixed